### PR TITLE
fix(snapshot): define resourceToPreview from resources changed. 

### PR DESCRIPTION
**Pull request by louis-bompart**
## Proposed changes

The current algorithm doesn't take into account the resources the user can access.
To solves that, we could either:
 - Query/cache what resources the user can access when we query it
 - Only do the expanded preview on resources that change.

## Testing

- [ ] Unit Tests:
Snapshot UTs are tangled, changing it will be difficult/require a rewrite.
- [ ] Functionnal Tests:
Corner-case, would require to create orgs without some access right or some stuff like that.
- [ ] Manual Tests:
In the only known affected environment (customer-owned), I managed to reproduce the issue the same customer reported.
 - Tested with the current version of the CLI, dev build from master: Crash
 - Tested with dev build from this branch: No Crash.

### CODE REVIEW

File: packages/cli/core/src/lib/snapshot/snapshotReporter.ts [hunk 1]
```diff
@@ -67,6 +67,30 @@ export class SnapshotReporter {
...
+    const isResourceTypeReportAffected = (
+      report: ResourceSnapshotsReportOperationModel
+    ) => {
+      for (const operation of Object.values(report)) {
+        if (operation > 0) {
+          return true;
+        }
+      }
+      return false;
+    };
```
Comment (olamothe) [inline packages/cli/core/src/lib/snapshot/snapshotReporter.ts:80]: Suggestion: 

```suggestion
    const isResourceTypeReportAffected = (
      report: ResourceSnapshotsReportOperationModel
    ) => {
    
    Object.values(report).reduce((operation, total) => operation + total, 0) === 0;
```

**Review by olamothe — APPROVED** [review]
